catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/SeedCommand.java

```diff
@@ -0,0 +1,195 @@
+                    + "products.")
+    int productLimit = 20;
+
+    @Reference
+    CatalogFramework framework;
+
+    @Reference
+    FilterBuilder filterBuilder;
+
+    private final Predicate<Metacard> isNonCachedResource = metacard -> {
+        Attribute cached = metacard.getAttribute(RESOURCE_CACHE_STATUS);
+        return cached == null || !((Boolean) cached.getValue());
+    };
+
+    private Filter createFilter() throws Exception {
+        if (isNotBlank(cql)) {
+            return CQL.toFilter(cql);
+        } else {
+            return filterBuilder.attribute(ANY_TEXT)
+                    .is()
+                    .like()
+                    .text(WILDCARD_CHAR);
+        }
+    }
+
+    @Override
+    public Object execute() throws Exception {
+        return doExecute();
+    }
+
+    @Override
+    protected Object executeWithSubject() throws Exception {
+        if (productLimit <= 0) {
+            printErrorMessage("A limit of " + productLimit
+                    + " was supplied. The limit must be greater than 0.");
+            return null;
+        }
+
+        final Filter filter = createFilter();
+
+        final long start = System.currentTimeMillis();
+        int productDownloads = 0;
+        int downloadErrors = 0;
+        int pageCount = 0;
+
+        while (productDownloads < productLimit) {
+            final QueryImpl query = new QueryImpl(filter);
+            query.setPageSize(productLimit);
+            query.setStartIndex(pageCount * productLimit + 1);
+            ++pageCount;
+
+            final QueryRequestImpl queryRequest;
+            if (sources != null && !sources.isEmpty()) {
+                queryRequest = new QueryRequestImpl(query, sources);
+            } else {
+                queryRequest = new QueryRequestImpl(query, true);
+            }
+            queryRequest.setProperties(new HashMap<>(CACHE_UPDATE_PROPERTIES));
+
+            final QueryResponse queryResponse = framework.query(queryRequest);
+
+            if (queryResponse.getResults()
+                    .isEmpty()) {
+                break;
+            }
+
+            final List<Entry<? extends ResourceRequest, String>> resourceRequests =
+                    queryResponse.getResults()
+                            .stream()
+                            .map(Result::getMetacard)
+                            .filter(isNonCachedResource)
+                            .map(metacard -> new SimpleEntry<>(new ResourceRequestById(metacard.getId()),
+                                    metacard.getSourceId()))
+                            .collect(toList());
+
+            for (Entry<? extends ResourceRequest, String> requestAndSourceId : resourceRequests) {
+                final ResourceRequest request = requestAndSourceId.getKey();
+                try {
+                    framework.getResource(request, requestAndSourceId.getValue());
```

Comment:
Unfortunately I think it will only work that way if `Always Cache Product` has been set in the resource download settings: https://github.com/codice/ddf/blob/master/catalog/core/catalog-core-standardframework/src/main/resources/OSGI-INF/metatype/downloadManager.xml#L43